rosbag2_cpp/test/rosbag2_cpp/test_time_controller_clock.cpp

```diff
@@ -112,3 +115,51 @@ TEST_F(TimeControllerClockTest, slow_rate)
   return_time = some_time;
   EXPECT_EQ(pclock.now(), as_nanos(some_time) * playback_rate);
 }
+
+TEST_F(TimeControllerClockTest, basic_pause_resume)
```

Comment:
I'm wondering if it's better to break this into multiple smaller tests. Generally with this pattern

```cpp
// Comment
...code...

// Comment
...code...

// Comment
...code...
```

it may be better written as separate tests that can run independently.

Reply:
OK - good idea